### ISSUE TITLE
groupBy shows group name but doesn't group results

### ISSUE THREAD

**Issue by LouNik1984** [devbridge/jQuery-Autocomplete#859]
I'm testing the groupBy option with the demo provided. The demo array "teams" is structured so to have all the items from the first category first, and then lists all the item from the second category:

`
   var nhlTeams = ['Anaheim Ducks', 'Atlanta Thrashers', 'Boston Bruins', ...];
   var nbaTeams = ['Atlanta Hawks', 'Boston Celtics', 'Charlotte Bobcats', ...];
   var nhl = $.map(nhlTeams, function (team) { return { value: team, data: { category: 'NHL' }}; });
   var nba = $.map(nbaTeams, function (team) { return { value: team, data: { category: 'NBA' } }; });
   var teams = nhl.concat(nba);
`

This way, when we perform a search using  the  groupBy option, the search results shows first all the results from the first category, then all the results from the second category. In this way, the results from the same category are grouped together:

<img width="337" alt="Screenshot 2024-10-02 alle 12 15 58" src="https://github.com/user-attachments/assets/3a8c2cf7-ed8d-429c-b9fb-388d58c5e81c">

If we shuffle the teams items in the array so that final results from both categories are mixed:

`
searchquery = "bl";

var teamsddd = [
        {"value": "Anaheim Ducks", "data": {"category": "NHL"}},
        {"value": "Portland Trail Blazers", "data": {"category": "NBA"}}, // this will show in search with category NBA
        {"value": "Atlanta Thrashers", "data": {"category": "NHL"}},
        {"value": "Chicago Blackhawks", "data": {"category": "NHL"}}, // this will show in search with category NHL
        {"value": "Boston BlCeltics", "data": {"category": "NBA"}}, // this will show in search with category NBA
        {"value": "Columbus Blue Jackets", "data": {"category": "NHL"}}, // this will show in search with category NHL
        {"value": "Detroit Red Wings", "data": {"category": "NHL"}},
        {"value": "Atlanta Hawks", "data": {"category": "NBA"}},
        {"value": "Cleveland Cavaliers", "data": {"category": "NBA"}},
        {"value": "Dallas Mavericks", "data": {"category": "NBA"}},
    ];
`
I expect to still see the results grouped by their categories, so I expect to see the first category with its two entries, and then the second category with the other two entries, no matter where they are in the array:

`
NBA
- portland trail blazers // array position 2
- boston Blceltics  // array position 5
NHL
- chicago blackhawks  // array position 4
- columbus blue jackets  // array position 6
`

What I get, instead, is the four items of the array in the order that are listed in the array:

<img width="382" alt="Screenshot 2024-10-02 alle 12 19 49" src="https://github.com/user-attachments/assets/4d91692a-ee43-475d-8707-6048b2ec7b50">

Basically, the grouping gets lost. Is this intended behaviour? To me, a groupBy option means that results should be grouped under that category depsite  their original position in  the array

**Comment by tkirda** [comment]
@LouNik1984 you are correct. This is a bug. Do you want to create a PR with the fix?

**Comment by LouNik1984** [comment]
@tkirda 

Hi

I don't have a fix  ready. If somebody can look into it and provide a PR with the  fix, it would be great

**Comment by tkirda** [comment]
Fixed in [85796a9](https://github.com/devbridge/jQuery-Autocomplete/commit/85796a9) — shipping in the next 2.x release.

The renderer now reorders suggestions by their `groupBy` field before drawing, so each category appears once with its matching items grouped beneath it. First-occurrence order across categories is preserved, and relative order within each category is preserved too. With your example, querying `bl` would now render:

```
NBA
- Portland Trail Blazers
- Boston BlCeltics
NHL
- Chicago Blackhawks
- Columbus Blue Jackets
```

Thanks for the clear repro — that made the fix straightforward.
